Modules/posixmodule.c

```diff
@@ -14462,9 +14463,11 @@ INITFUNC(void)
         }
 
         /* initialize scandir types */
-        if (PyType_Ready(&ScandirIteratorType) < 0)
+        ScandirIteratorType = (PyTypeObject *)PyType_FromSpec(&ScandirIteratorType_spec);
```

Comment:
This assumes the `INITFUNC` is run exactly once per process. The classes will be shared across subinterpreters, losing most benefits of heap types.